src/thunderbots/software/network_input/networking/network_client.cpp

```diff
@@ -90,12 +90,20 @@ void NetworkClient::filterAndPublishVisionData(SSL_WrapperPacket packet)
             ret.first->second = detection;
         }
 
+        std::optional<Timestamp> latest_timestamp = std::nullopt;
```

Comment:
I think latest_detections could be confused as either 'the newest' or 'a delayed/old detections'.

I think what you're going for is 'newest_detections"?

Reply:
Ultimately removed this variable so this is resolved